apps/src/required_block_utils.js

```diff
@@ -146,6 +151,43 @@ function testsFromProcedure(node) {
   }];
 }
 
+function testsFromFunctionalCall(node, blocksXml) {
+  var name = node.querySelector('mutation').getAttribute('name');
+  var argElements = node.querySelectorAll('arg');
+  var types = [];
+  for (var i = 0; i < argElements.length; i++) {
+    types.push(argElements[i].getAttribute('type'));
+  }
+
+  var definition = _.find(blocksXml.childNodes, function (sibling) {
+    if (sibling.getAttribute('type') !== 'functional_definition') {
+      return false;
+    }
+    var nameElement = sibling.querySelector('title[name="NAME"]');
+    if (!nameElement) {
+      return false;
+    }
+    return nameElement.textContent === name;
+  });
+
+  if (!definition) {
+    throw new Error('No matching definition for functional_call');
+  }
+
+  return [{
+    test: function (userBlock) {
+      if (userBlock.type !== 'functional_call' ||
+          userBlock.getCallName() !== name) {
+        return false;
+      }
+      var userTypes = userBlock.getParamTypes();
+      return _.isEqual(userTypes, types);
+    },
+    blockDisplayXML: xml.serialize(definition) + xml.serialize(node)
```

Comment:
Fair point. In theory, there might be edges cases like that. In practice, we should never be having complex required blocks so I'm not too concerned and don't want to spend a bunch of time over engineering this.